### PR TITLE
feat: deeplink-schema-match typescript

### PR DESCRIPTION
**Pull request by limpbrains**

### CODE REVIEW

**Comment by GladosBlueWallet** [comment]
♫ This was a triumph. I'm making a note here: HUGE SUCCESS ♫

 [android in browser] https://appetize.io/app/xhfib722ldzdt3htkchpwvrk3i?device=pixel4

[download apk](https://lambda-download-android-build.herokuapp.com/download/433)

**Comment by GladosBlueWallet** [comment]
♫ This was a triumph. I'm making a note here: HUGE SUCCESS ♫

 [android in browser] https://appetize.io/app/xbmgqrq6qpcgedpehiwf732lka?device=pixel4

[download apk](https://lambda-download-android-build.herokuapp.com/download/435)

**Comment by GladosBlueWallet** [comment]
♫ This was a triumph. I'm making a note here: HUGE SUCCESS ♫

 [android in browser] https://appetize.io/app/p7tlbk6s4qyykdoyv4nozbxd5a?device=pixel4

[download apk](https://lambda-download-android-build.herokuapp.com/download/443)

**Comment by GladosBlueWallet** [comment]
Wake the fuck up samurai, we have PRs to merge

image

[all PRs for @Overtorment] https://github.com/BlueWallet/BlueWallet/pulls/review-requested/Overtorment

**Comment by GladosBlueWallet** [comment]
Wake the fuck up samurai, we have PRs to merge

image

[all PRs for @Overtorment] https://github.com/BlueWallet/BlueWallet/pulls/review-requested/Overtorment

**Comment by GladosBlueWallet** [comment]
Wake the fuck up samurai, we have PRs to merge

image

[all PRs for @Overtorment] https://github.com/BlueWallet/BlueWallet/pulls/review-requested/Overtorment

**Comment by GladosBlueWallet** [comment]
Wake the fuck up samurai, we have PRs to merge

image

[all PRs for @Overtorment] https://github.com/BlueWallet/BlueWallet/pulls/review-requested/Overtorment

File: typings/bip21.d.ts [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+    | {
+        amount?: number;
+        label?: string;
+        pj?: string;
+      }
+    | { [key: string]: string };
+
+  export function decode(uri: string, urnScheme?: string): { address: string; options: TOptions };
+  export function encode(address: string, options?: TOptions, urnScheme?: string): string;
+}
```
Comment (Overtorment) [inline typings/bip21.d.ts:12]: very noice! btw can this be published to npm so anyone  can install typings for bip21 package..?

File: class/deeplink-schema-match.ts [hunk 2]
```diff
@@ -450,8 +462,8 @@ class DeeplinkSchemaMatch {
     return undefined;
   }
 
-  static bip21decode(uri) {
-    if (!uri) return {};
+  static bip21decode(uri: string) {
+    // if (!uri) return {};
```
Comment (Overtorment) [inline class/deeplink-schema-match.ts:466]: hmm, this guard clause was here for a reason, probably

Comment (Overtorment) [reply class/deeplink-schema-match.ts:466]: was  added here:  https://github.com/BlueWallet/BlueWallet/commit/6565fd531dab5b8757061151e615ed0fcb4bd709

**Review by Overtorment — CHANGES_REQUESTED** [review]
lgtm, but this commented out line bothers me

File: typings/bip21.d.ts (hunk 1 above)
Comment (limpbrains) [reply typings/bip21.d.ts:12]: IT is typed, but not good enought, doesn't export Options, doesn't have `pj`
I don't wan't to bother with it
https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/bip21/index.d.ts

File: class/deeplink-schema-match.ts (hunk 2 above)
Comment (limpbrains) [reply class/deeplink-schema-match.ts:466]: I'm not a fan of "return something simular" in case of error. I would rather throw an error in this case

**Comment by GladosBlueWallet** [comment]
♫ This was a triumph. I'm making a note here: HUGE SUCCESS ♫

 [android in browser] https://appetize.io/app/neeq5du2n3zrnk5rc4y2fyrdyy?device=pixel4

[download apk](https://lambda-download-android-build.herokuapp.com/download/471)

**Comment by GladosBlueWallet** [comment]
♫ This was a triumph. I'm making a note here: HUGE SUCCESS ♫

 [android in browser] https://appetize.io/app/xix6cdvsvbjwuqrsif5niouesy?device=pixel4

[download apk](https://lambda-download-android-build.herokuapp.com/download/474)

**Review by Overtorment — APPROVED** [review]

**Comment by GladosBlueWallet** [comment]
♫ This was a triumph. I'm making a note here: HUGE SUCCESS ♫

 [android in browser] https://appetize.io/app/w4gqdkpcsnh7g535s5ef2yvxw4?device=pixel4

[download apk](https://lambda-download-android-build.herokuapp.com/download/491)

**Comment by limpbrains** [comment]
please, sir, can I merge?

**Comment by GladosBlueWallet** [comment]
♫ This was a triumph. I'm making a note here: HUGE SUCCESS ♫

 [android in browser] https://appetize.io/app/ii64ajrrwtwdzkzkwmrx4b5cai?device=pixel4

[download apk](https://lambda-download-android-build.herokuapp.com/download/498)

**Comment by Overtorment** [comment]
glados, wake up

**Comment by GladosBlueWallet** [comment]
Unbelievable. You, [subject name here], must be the pride of [subject hometown here]!

**Comment by Overtorment** [comment]
fizxed glados in https://github.com/BlueWallet/GlaDOS/commit/bc369dda5f9390e36b4f6baa854249afea2f0a5b

**Pull request merged**